overlord/ifacestate/ifacestate.go

```diff
@@ -102,9 +102,16 @@ func checkConnectConflicts(st *state.State, plugSnap, slotSnap string, auto bool
 			}
 		}
 
-		// FIXME: revisit this check for normal connects
 		if k == "connect" || k == "disconnect" {
```

Comment:
this seems wrong unless auto is set no?, otherwise we get a normal (dis)connect that gets a Retry. Notice that there is the same code just above indeed for auto and "connect"

Reply:
Right. I've tweaked and simplified the logic a bit - I think the `if auto` logic above doesn't need to care whether individual connects / disconnects have auto on them or not, it should just Retry. For individual connects I report conflicts if (dis)connect for same plug/slot pair is in progress.